service/storage_service.cc

```diff
@@ -1835,15 +1840,104 @@ future<> storage_service::decommission() {
             }
 
             slogger.info("DECOMMISSIONING: starts");
-            ss.start_leaving().get();
-            // FIXME: long timeout = Math.max(RING_DELAY, BatchlogManager.instance.getBatchlogTimeout());
-            auto timeout = ss.get_ring_delay();
-            ss.set_mode(mode::LEAVING, format("sleeping {} ms for batch processing and pending range setup", timeout.count()), true);
-            sleep_abortable(timeout, ss._abort_source).get();
+            auto leaving_nodes = std::list<gms::inet_address>{endpoint};
+            std::list<gms::inet_address> ignore_nodes;
+
+            future<> heartbeat_updater = make_ready_future<>();
+            auto heartbeat_updater_done = make_lw_shared<bool>(false);
```

Comment:
why do we need `heartbeat_updater_done` to be lw_shared?

Reply:
because it is easier and safer to share between those two fibers.